api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto

```diff
@@ -173,6 +174,9 @@ message ExternalProcessor {
   // :ref:`clear_route_cache <envoy_v3_api_field_service.ext_proc.v3.CommonResponse.clear_route_cache>`
   // field is set in an external processor response.
   bool disable_clear_route_cache = 11;
+
+  // Disallow headers to be forwarded to the ext_proc server based on this regular expression.
+  type.matcher.v3.RegexMatcher disallow_header_to_forward = 12;
```

Comment:
@htuch @rshriram @tyxia 
To have ext_proc filter support header scrubbing in the direction from envoy to ext_proc server, we need to add this regex expression configuration. Thus ext_proc filter can remove headers based on this config.

On the reverse direction,  Envoy has this configure for this purpose: 

https://github.com/envoyproxy/envoy/blob/23168f3cc7ea9693e12cfda065268c5e4957c570/api/envoy/config/common/mutation_rules/v3/mutation_rules.proto#L79

The question I have is that is it enough to just add this disallow_header_to_forward regex expression?  Do we need to add something like:
config.common.mutation_rules.v3.HeaderMutationRules outbound_mutation_rules = 12;

similar to the reverse direction:

https://github.com/envoyproxy/envoy/blob/23168f3cc7ea9693e12cfda065268c5e4957c570/api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto#L165
 
which has a lot of other capabilities, which we don't use it for now, but may be needed in the future?

Reply:
IMHO, a wrapper seems to be better.  it will be flexible and extensible with more feature/control and without compatibility concern (e.g., move this field into wrapper) or add more fields scatteredly 

`HeaderMutationRules` in mutation_rules.proto is specific to mutation by external processing server as all the code comment suggests. Since we don't need all those sub fields there and our purpose here is different (i.e., forward not mutation), I think it will be clear that we introduce a new wrapper e.g., `HeaderForwardingRules` in mutation_rules.proto. It is fine that we just have single regex field for now.

WDYT?

Reply:
We need an allowlist. Like headers_to_forward. Its a simple list of headers. This will be specified by the end user. Disallow may be too cumbersome.

Reply:
+1. It should be an allow list. My preference TBH is to make this a `repeated Regex`. I don't see a lot of value in forcing all users to form a compound regex to express the common case of a list of headers. Internally we can still turn it into one large regex if we want to for efficiency or the like.

Reply:
Good ideas for all the comments! Just updated the PR based on them.